app/kubernetes-ingress-controller/2.0.x/guides/using-udpingress.md

```diff
@@ -0,0 +1,323 @@
+mechanism to automatically enable new UDP ports when you need them to expose your
+Kubernetes UDP `Services`, so we will need to explicitly configure Kong to expose
+these ports prior to deploying our `UDPIngress` resources.
+
+If you're maintaining a `values.yaml` configuration for your Chart deployment of Kong
+you will need to add a section under `udpProxy` which enables the new UDP listener:
+
+```yaml
+udpProxy:
+  enabled: true
+  type: LoadBalancer
+  stream:
+  - containerPort: 9999
+    servicePort: 9999
+    protocol: UDP
+```
+
+Once you've made the necessary configurations you can apply your changes:
+
+```
+$ helm upgrade --namespace <NAMESPACE> --version <CHART_VERSION> -f values.yaml <RELEASE_NAME> kong/kong
+```
+
+Replace `<NAMESPACE>`, `<CHART_VERSION>` and `<RELEASE_NAME>` with the values you deployed Kong with.
+
+Alternatively if you are using only command line flags to deploy and manage Kong
+the same configuration can be achieved with flags:
+
+```shell
+$ helm upgrade --namespace <NAMESPACE> --version <CHART_VERSION> <RELEASE_NAME> kong/kong \
+  --set "udpProxy.enabled=true" \
+  --set "udpProxy.type=LoadBalancer" \
+  --set "udpProxy.stream[0].containerPort=9999"
+  --set "udpProxy.stream[0].servicePort=9999"
+  --set "udpProxy.stream[0].protocol=UDP"
+```
+
+Wait for the `LoadBalancer` service to be ready for the Gateway (see `kubectl get services`)
+and once it's up Kong is ready to serve UDP traffic on port `9999` external to the cluster
+using `UDPIngress` resources.
+
+## Deploying UDPIngress
+
+Now that Kong is listening on `9999` we can create a `UDPIngress` resource which will attach
+our CoreDNS service to that port so we can make DNS requests to it from outside the cluster.
+
+```yaml
+apiVersion: configuration.konghq.com/v1beta1
+kind: UDPIngress
+metadata:
+  name: minudp
+  namespace: udpingress-example
+  annotations:
+    kubernetes.io/ingress.class: kong
+spec:
+  rules:
+  - backend:
+      serviceName: coredns
+      servicePort: 53
+    port: 9999
+```
+
+The above binds the Kong Gateway port `9999` to the `Service` port `53` for our DNS server.
+
+Save the file as `coredns-udpingress.yaml` and apply it:
+
+```shell
+$ kubectl apply -f coredns-udpingress.yaml
+```
+
+We can now make DNS requests via Kong.
+
+## Verification
+
+The exercise is complete and all that's left to do is verify that everything is working
+by making a DNS request to our CoreDNS server.
+
+This example assumes you have the `dig` command available on your local system, but if
+you don't look to your operating system documentation for a similar DNS lookup tool.
+
+Firstly you'll need to retrieve the IP address of the UDP load balancer service that we
+configured in previous sections:
+
+```shell
+$ export KONG_UDP_ENDPOINT="$(kubectl -n <NAMESPACE> get service <RELEASE_NAME>-kong-udp-proxy \
+    -o=go-template='{{range .status.loadBalancer.ingress}}{{.ip}}{{end}}')"
```

Comment:
```suggestion
$ export KONG_UDP_ENDPOINT="$(kubectl -n {NAMESPACE} get service {RELEASE_NAME}-kong-udp-proxy \
    -o=go-template='{% raw %}{{range .status.loadBalancer.ingress}}{{.ip}}{{end}}{% endraw %}')"
```
In Jekyll, `{{ }}` refer to variables, so it's trying to parse the tag and not getting anything. These `{% raw %}` tags tell Jekyll to read them as plaintext instead, which is what we want so they display properly.

Here's what it's currently generating:
![Screen Shot 2021-08-23 at 2 03 45 PM](https://user-images.githubusercontent.com/54370747/130519158-8b6ce691-6c32-44c8-af05-67b6d0547266.png)